apps/aecore/src/aec_dry_run.erl

```diff
@@ -15,8 +15,11 @@
 dry_run(TopHash, Accounts, Txs) ->
     try setup_dry_run(TopHash, Accounts) of
         {Env, Trees} -> dry_run_(Txs, Trees, Env)
-    catch _E:_R ->
-        {error, <<"Failed to get block state and environment">>}
+    catch
```

Comment:
This looks weird... why would the refactoring change the behaviour so much?

Reply:
The PR introduces new behavior (before we had all old states...) so it is nice if that is signalled to the user when he tries to use a state that is garbage collected!

Reply:
It's improvement in a way that we can differentiate between block hash not in DB, and situation where block hash is in DB, but state for that block points to garbage collected root of MP-trie.